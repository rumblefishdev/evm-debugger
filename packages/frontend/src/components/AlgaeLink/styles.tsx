import type { LinkProps } from '@mui/material';
import { Link, styled } from '@mui/material';

export const StyledSpan = styled('span')(() => ({
  textDecoration: 'none',
  display: 'block',
})) as React.FC<LinkProps>;

export const StyledLink = styled(Link)(() => ({
  textDecoration: 'none',
})) as React.FC<LinkProps>;
