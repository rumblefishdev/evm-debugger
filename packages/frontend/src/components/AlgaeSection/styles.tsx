import { Container, styled } from '@mui/material'

import { sectionClasses } from './Section.types'

export const StyledRoot = styled(Container)(({ theme }) => ({
  [`&.${sectionClasses.full}`]: {
    width: '100%',
    maxWidth: '100%',
  },
  [`&.${sectionClasses.normal}`]: {
    maxWidth: '1280px',
    [theme.breakpoints.up('md')]: {
      width: '95%',
    },
  },
  [`&.${sectionClasses.small}`]: {
    maxWidth: '1230px',
    [theme.breakpoints.up('md')]: {
      width: '95%',
    },
  },

  [`&.${sectionClasses.mobilePadding}`]: {
    ...theme.mixins.mobilePadding(),
  },
}))
