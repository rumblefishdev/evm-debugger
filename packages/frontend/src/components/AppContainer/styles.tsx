import { Stack, styled } from '@mui/material'

export const StyledWrapper = styled(Stack)(() => ({
  justifyContent: 'center',
  alignItems: 'center',
}))

export const StyledContainer = styled(Stack)(() => ({
  width: '100%',
  overflow: 'hidden',
  maxWidth: '1600px',
  justifyContent: 'center',
  height: '100vh',
  alignItems: 'center',
}))
