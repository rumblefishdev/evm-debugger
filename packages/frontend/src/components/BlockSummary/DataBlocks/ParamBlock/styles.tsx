import { Accordion, Stack, styled, Typography } from '@mui/material'

export const StyledAccordion = styled(Accordion)(() => ({}))
