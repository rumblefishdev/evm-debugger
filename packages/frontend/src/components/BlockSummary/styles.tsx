import { Card, Stack, styled, Typography } from '@mui/material'

export const StyledBox = styled(Card)(() => ({
  width: '100%',
  padding: '24px',
  maxWidth: '824px',
  height: '100%',
  borderRadius: '18px',
}))

export const StyledWrapper = styled(Stack)(() => ({}))

export const StyledInfoRow = styled(Stack)(() => ({
  width: '100%',
  marginBottom: '8px',
  flexDirection: 'row',
}))

export const StyledInfoType = styled(Typography)(() => ({
  width: '128px',
}))

export const StyledInfoValue = styled(Typography)(() => ({
  width: 'cacl(100% - 128px - 24px)',
  overflowWrap: 'anywhere',
}))
