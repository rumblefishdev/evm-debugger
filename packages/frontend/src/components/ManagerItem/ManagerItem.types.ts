import type { StackProps } from '@mui/material'

export interface ManagerItemProps extends StackProps {
  name: string
  value: string
  isFound: boolean
  address?: string
  updateItem: (id: string, value: string) => void
}
