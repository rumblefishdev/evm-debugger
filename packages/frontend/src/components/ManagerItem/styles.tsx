import { Stack, styled, Typography } from '@mui/material'

const StyledStatus = styled(Typography)(({ theme }) => ({
  padding: theme.spacing(1, 2),
  ...theme.typography.caption,
  width: '100%',
  maxWidth: '80px',
  color: theme.palette.rfWhite,
  borderRadius: '21px',
}))

export const StyledStatusFound = styled(StyledStatus)(({ theme }) => ({
  backgroundColor: theme.palette.rfSuccess,
}))

export const StyledStatusNotFound = styled(StyledStatus)(({ theme }) => ({
  backgroundColor: theme.palette.rfBrandSecondary,
}))

export const StyledStack = styled(Stack)(({ theme }) => ({
  width: '100%',
  padding: theme.spacing(3.5, 4),
  maxWidth: '764px',
  justifyContent: 'flex-start',
  gap: theme.spacing(4),
  flexDirection: 'row',
  boxSizing: 'border-box',
  border: `1px solid ${theme.palette.rfLinesLight}`,
  backgroundColor: theme.palette.rfBackground,
  alignItems: 'center',
}))

export const StyledName = styled(Typography)(({ theme }) => ({
  width: '100%',
  maxWidth: '585px',
  ...theme.typography.bodySmall,
  color: theme.palette.rfText,
}))
