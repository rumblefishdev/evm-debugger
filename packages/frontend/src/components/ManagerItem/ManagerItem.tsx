import { useState } from 'react'
import { Tooltip } from '@mui/material'

import { Button } from '../Button'
import { DataAdder } from '../DataAdder'
import { TransactionContentDialog } from '../TransactionContentDialog'

import type { ManagerItemProps } from './ManagerItem.types'
import { StyledStack, StyledName, StyledStatusFound, StyledStatusNotFound } from './styles'

export const ManagerItem = ({ isFound, address, name, value, updateItem, contentType, ...props }: ManagerItemProps) => {
  const [isDataVisible, setDataVisibility] = useState(false)
  const [isDataAdderVisible, setDataAdderVisibility] = useState(false)

  const dataAdderHandler = (data: string) => {
    updateItem(name, data)
    setDataAdderVisibility(false)
  }

  return (
    <StyledStack {...props}>
      {isFound ? <StyledStatusFound>Found</StyledStatusFound> : <StyledStatusNotFound>Not found</StyledStatusNotFound>}
      <Tooltip
        title={address}
        arrow
        followCursor
      >
        <StyledName>{name || address}</StyledName>
      </Tooltip>
      <Button
        variant="text"
        size="medium"
        onClick={isFound ? () => setDataVisibility(true) : () => setDataAdderVisibility(true)}
      >
        {isFound ? 'Show' : 'Add'}
      </Button>
      {isFound ? (
        <TransactionContentDialog
          title={name}
          description={address}
          content={value}
          contentType={contentType}
          open={isDataVisible}
          onClose={() => setDataVisibility(false)}
        />
      ) : (
        <DataAdder
          open={isDataAdderVisible}
          submithandler={dataAdderHandler}
          title={`Upload for ${name}`}
          onClose={() => setDataAdderVisibility(false)}
          description={address}
        />
      )}
    </StyledStack>
  )
}
