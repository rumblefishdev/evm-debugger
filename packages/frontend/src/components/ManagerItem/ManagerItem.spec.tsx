import * as React from 'react'
import renderer from 'react-test-renderer'

import { ManagerItem } from './ManagerItem'
import { StyledName } from './styles'

describe('ManagerItem tests', () => {
  it('simple render', () => {
    const expectedText = 'sampleName'
    const render = renderer.create(
      <ManagerItem
        key="0x1"
        address="0x2"
        name={expectedText}
        value="sampleValue"
        isFound={false}
        updateItem={null}
      />,
    )
    const testInstance = render.root.findByType(StyledName)
    expect(testInstance.props.children).toEqual(expectedText)
    console.log(render.toJSON())
    expect(render.toJSON()).toMatchSnapshot()
  })

  it('display address if name not exist', () => {
    const expectedText = '0x2'
    const render = renderer.create(
      <ManagerItem
        key="0x1"
        address={expectedText}
        value="sampleValue"
        isFound={false}
        updateItem={null}
        name={null}
      />,
    )
    const testInstance = render.root.findByType(StyledName)
    expect(testInstance.props.children).toEqual(expectedText)
  })
})
