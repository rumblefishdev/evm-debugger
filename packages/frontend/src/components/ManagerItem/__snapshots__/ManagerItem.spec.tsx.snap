// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManagerItem tests simple render 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-5bbpg0{width:100%;padding:28px 32px;max-width:764px;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;gap:32px;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;box-sizing:border-box;border:1px solid undefined;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
      }
    }
    data-emotion="css 5bbpg0"
  />,
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1xwjwa8-MuiStack-root{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-flex-direction:column;-ms-flex-direction:column;flex-direction:column;width:100%;padding:28px 32px;max-width:764px;-webkit-box-pack:start;-ms-flex-pack:start;-webkit-justify-content:flex-start;justify-content:flex-start;gap:32px;-webkit-flex-direction:row;-ms-flex-direction:row;flex-direction:row;box-sizing:border-box;border:1px solid undefined;-webkit-align-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;}",
      }
    }
    data-emotion="css 1xwjwa8-MuiStack-root"
  />,
  <div
    className="css-1xwjwa8-MuiStack-root"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-3hfi54{padding:8px 16px;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.75rem;line-height:1.66;letter-spacing:0.03333em;width:100%;max-width:80px;border-radius:21px;}",
        }
      }
      data-emotion="css 3hfi54"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-amio4c-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;padding:8px 16px;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:0.75rem;line-height:1.66;letter-spacing:0.03333em;width:100%;max-width:80px;border-radius:21px;}",
        }
      }
      data-emotion="css amio4c-MuiTypography-root"
    />
    <p
      className="MuiTypography-root MuiTypography-body1 css-amio4c-MuiTypography-root"
    >
      Not found
    </p>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1us7b4i{width:100%;max-width:585px;}",
        }
      }
      data-emotion="css 1us7b4i"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-9tbx7m-MuiTypography-root{margin:0;font-family:"Roboto","Helvetica","Arial",sans-serif;font-weight:400;font-size:1rem;line-height:1.5;letter-spacing:0.00938em;width:100%;max-width:585px;}",
        }
      }
      data-emotion="css 9tbx7m-MuiTypography-root"
    />
    <p
      aria-label="0x2"
      aria-labelledby={null}
      className="MuiTypography-root MuiTypography-body1  css-9tbx7m-MuiTypography-root"
      data-mui-internal-clone-element={true}
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
    >
      sampleName
    </p>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-pnutvq-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-pnutvq-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css pnutvq-MuiTooltip-popper"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-viryjt-MuiPopper-root-MuiTooltip-popper{z-index:1500;pointer-events:none;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow{top:0;margin-top:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="bottom"] .MuiTooltip-arrow::before{transform-origin:0 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow{bottom:0;margin-bottom:-0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="top"] .MuiTooltip-arrow::before{transform-origin:100% 0;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow{left:0;margin-left:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="right"] .MuiTooltip-arrow::before{transform-origin:100% 100%;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow{right:0;margin-right:-0.71em;height:1em;width:0.71em;}.css-viryjt-MuiPopper-root-MuiTooltip-popper[data-popper-placement*="left"] .MuiTooltip-arrow::before{transform-origin:0 0;}",
        }
      }
      data-emotion="css viryjt-MuiPopper-root-MuiTooltip-popper"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1dkn99d{outline:none;cursor:pointer;border:none;font-weight:600;font-size:14px;font-family:Rajdhani;background-color:unset;}",
        }
      }
      data-emotion="css 1dkn99d"
    />
    <button
      className="MuiButton-root css-1dkn99d"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      sx={{}}
      variant="text"
    >
      Add
    </button>
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-1wdzqm1{background:rgba(1,25,111,0.9);}",
        }
      }
      data-emotion="css 1wdzqm1"
    />
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-dqe6ny-MuiDialog-root{background:rgba(1,25,111,0.9);}@media print{.css-dqe6ny-MuiDialog-root{position:absolute!important;}}",
        }
      }
      data-emotion="css dqe6ny-MuiDialog-root"
    />
  </div>,
]
`;
